[8.16] Adds entity store docs (#6053)
elastic/security-docs: docs/advanced-entity-analytics/entity-store.asciidoc

[[entity-store]]
= Entity store

preview::[]

.Requirements
[sidebar]
--
To use the entity store, you must have the appropriate privileges. For more information, refer to <<ers-requirements, Entity risk scoring requirements>>.
--

> The entity store allows you to query, reconcile, maintain, and persist entity metadata from various sources such as:
> 
> * Ingested log sources
> * Integrated identity providers (such as Active Directory, EntraID, and Okta)
> * Internal and external alerts
> * External asset repositories
> * Asset criticality submissions

Reviewer: ```suggestion
The entity store allows you to query, reconcile, maintain, and persist entity metadata such as:

* Ingested logs
* Integrated identity data (such as from Active Directory, EntraID, and Okta)
* Internal and external alerts
* External asset repository data
* Asset criticality data
```
Just another option that occurred to me

Reviewer 2: We can also add "Entity risk score data" to this list

> The entity store can store any entity type observed by {elastic-sec}. This allows you to store, view, and query select entities directly from an index without needing to perform real-time searches of observable data. The entity store extracts entities from all indices in the {elastic-sec} <<default-data-view-security, default data view>>.

Reviewer: ```suggestion
The entity store can include any entity type observed by {elastic-sec}. It allows you to view and query select entities represented in your indices without needing to perform real-time searches of observable data. The entity store extracts entities from all indices in the {elastic-sec} <<default-data-view-security, default data view>>.
```
A suggestion that uses the word "store" a bit less. Not sure my suggestion is accurate but I thought that might be good since "store" hasn't really been defined in this context yet.

> When the entity store is enabled, the following resources are generated:

Reviewer: ```suggestion
When the entity store is enabled, the following resources are generated for each stored entity:
```
Not sure if this is accurate but might be nice to specify what the generated resources are for

Reviewer 2: I like the general suggestion, instead of "for each stored entity" we can instead say something like "for hosts and users", as the resources are created for each entity _type_, not each individual entity

* {es} resources, such as transforms, ingest pipelines, and enrich policies.
* Data and fields for each entity.
* The `.entities.v1.latest.security_user_<space-id>` and `.entities.v1.latest.security_host_<space-id>` indices, which contain field mappings for hosts and users respectively. You can query these indices to see a list of fields that are mapped in the entity store.

[discrete]
[[enable-entity-store]]
== Enable entity store

To enable the entity store:

. Find **Entity Store** in the navigation menu or by using the {kibana-ref}/introduction.html#kibana-navigation-search[global search field].
. On the **Entity Store** page, turn the toggle on. 

Once you enable the entity store, the Entity Analytics dashboard displays the <<entity-entities, **Entities**>> section.

[discrete]
[[clear-entity-store]]
== Clear entity store data

> Once the entity store is enabled, you may want to start fresh and clear the extracted entity data. For example, if you have refined your data by normalizing the `user.name` or `host.name` fields, clearing the entity store data allows you to repopulate the entity store with the updated, normalized values. This action removes all previously extracted entity information, enabling new data extraction and analysis.

Reviewer: ```suggestion
Once the entity store is enabled, you may want to clear existing data and start fresh. For example, if you normalized the `user.name` or `host.name` fields, clearing the entity store data would allow you to repopulate the entity store with the updated, normalized values. This action removes all previously extracted entity information, enabling new data extraction and analysis.
```

Clearing entity store data does not delete your source data, assigned entity risk scores, or asset criticality assignments.

> CAUTION: Clearing entity store data permanently deletes persisted user and host records, and data is no longer available for analysis. Proceed with caution, as this cannot be undone.

Reviewer 2: I kind of like the use of the word "persisted" here, to disambiguate from source data (as mentioned on line 45). I don't have strong opinions one way or the other though!
